di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/vc/VerifiableCredentialGenerator.java

```diff
@@ -0,0 +1,122 @@
+
+    public static final String EC_ALGO = "EC";
+    public static final String VC_ISSUER_ENV_VAR = "VC_ISSUER";
+    public static final String VC_SIGNING_KEY_ENV_VAR = "VC_SIGNING_KEY";
+
+    public SignedJWT generate(Credential credential, String subject)
+            throws NoSuchAlgorithmException, InvalidKeySpecException, JOSEException {
+
+        Map<String, Object> vc = new LinkedHashMap<>();
+        vc.put(VC_CONTEXT, new String[] {W3_BASE_CONTEXT, DI_CONTEXT});
+        vc.put(VC_TYPE, new String[] {VERIFIABLE_CREDENTIAL_TYPE, IDENTITY_CHECK_CREDENTIAL_TYPE});
+
+        Map<String, Object> credentialSubject = new LinkedHashMap<>();
+
+        Map<String, Object> attributes = new HashMap<>(credential.getAttributes());
+
+        if (isPopulatedList(attributes, CREDENTIAL_SUBJECT_NAME)) {
+            credentialSubject.put(CREDENTIAL_SUBJECT_NAME, attributes.get(CREDENTIAL_SUBJECT_NAME));
+            attributes.remove(CREDENTIAL_SUBJECT_NAME);
+        }
+
+        if (isPopulatedList(attributes, CREDENTIAL_SUBJECT_BIRTH_DATE)) {
+            credentialSubject.put(
+                    CREDENTIAL_SUBJECT_BIRTH_DATE, attributes.get(CREDENTIAL_SUBJECT_BIRTH_DATE));
+            attributes.remove(CREDENTIAL_SUBJECT_BIRTH_DATE);
+        }
+
+        if (isPopulatedList(attributes, CREDENTIAL_SUBJECT_ADDRESS)) {
+            credentialSubject.put(
+                    CREDENTIAL_SUBJECT_ADDRESS, attributes.get(CREDENTIAL_SUBJECT_ADDRESS));
+            attributes.remove(CREDENTIAL_SUBJECT_ADDRESS);
+        }
+        // Copy any remaining attributes in. The JSON manually entered into the stub.
+        credentialSubject.putAll(attributes);
+        vc.put(VC_CREDENTIAL_SUBJECT, credentialSubject);
+        // The schema is unclear on how this should be presented so just copying wholesale for now.
+        vc.put(VC_EVIDENCE, List.of(credential.getGpg45Score()));
+
+        return generateAndSignVerifiableCredentialJwt(subject, vc);
+    }
+
+    private boolean isPopulatedList(Map<String, Object> attributes, String attributeName) {
+        if (attributes.get(attributeName) instanceof List
+                && !((List<?>) attributes.get(attributeName)).isEmpty()) {
+            return true;
+        } else {
+            // in case of empty list
+            attributes.remove(attributeName);
+            return false;
+        }
+    }
+
+    private SignedJWT generateAndSignVerifiableCredentialJwt(String subject, Map<String, Object> vc)
+            throws NoSuchAlgorithmException, InvalidKeySpecException, JOSEException {
+        Instant now = Instant.now();
+        JWTClaimsSet claimsSet =
+                new JWTClaimsSet.Builder()
+                        .claim(SUBJECT, subject)
+                        .claim(ISSUER, System.getenv(VC_ISSUER_ENV_VAR))
+                        .claim(NOT_BEFORE, now.getEpochSecond())
+                        .claim(EXPIRATION_TIME, now.plusSeconds(60 * 5).getEpochSecond())
```

Comment:
Is this going to be enough time for the VC? Perhaps it could be an env var so that we can control this?

Reply:
Ah yeah, that’s a good shout. George thought 5 mins would be ok for the real thing, but it would be good for it to be configurable.

Reply:
@dannyhayes-gds Have pushed a commit to grab a value from the env (as well as moving other config stuff to the existing config class). There's also a PR here to set a value in the env from the manifest: https://github.com/alphagov/di-ipv-config/pull/207